src/azure-cli-telemetry/azure/cli/telemetry/__init__.py

```diff
@@ -46,10 +48,34 @@ def _start(config_dir):
 
 def save(config_dir, payload):
     from azure.cli.telemetry.util import should_upload
+    from azure.cli.telemetry.components.telemetry_client import CliTelemetryClient
     from azure.cli.telemetry.components.telemetry_logging import get_logger
 
-    if save_payload(config_dir, payload) and should_upload(config_dir):
-        logger = get_logger('main')
+    logger = get_logger('main')
+    try:
+        # Split payload to cli events and extra events by instrumentation key
+        # cli events should be stored in local cache and sent together
+        # extra events can be sent immediately
+        import json
+        events = json.loads(payload)
+
+        logger.info('Begin splitting cli events and extra events, total events: %s', len(events))
+        cli_events = {}
+        client = CliTelemetryClient()
+        for key, event in events.items():
+            if key == DEFAULT_INSTRUMENTATION_KEY:
+                cli_events[key] = event
+            else:
+                extra_event = {key: event}
+                client.add(json.dumps(extra_event), flush=True)
+        client.flush(force=True)
```

Comment:
Do we also need to set `client.flush(force=True)` when using the `DEFAULT_INSTRUMENTATION_KEY`?

Reply:
Yes we already have in https://github.com/Azure/azure-cli/blob/48a69dbebb15844afa9e240051a1c4d13ffa6bbb/src/azure-cli-telemetry/azure/cli/telemetry/__init__.py#L83-L86

Reply:
OK